src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/StorePal.Windows.Import.cs

```diff
@@ -118,12 +118,15 @@ private static StorePal FromBlobOrFile(ReadOnlySpan<byte> rawData, string? fileN
                 Interop.Crypt32.CertStoreFlags.CERT_STORE_ENUM_ARCHIVED_FLAG | Interop.Crypt32.CertStoreFlags.CERT_STORE_CREATE_NEW_FLAG | Interop.Crypt32.CertStoreFlags.CERT_STORE_DEFER_CLOSE_UNTIL_LAST_FREE_FLAG,
                 null);
 
-            if (certStore.IsInvalid ||
-                !Interop.Crypt32.CertAddCertificateLinkToStore(certStore, certificatePal.CertContext, Interop.Crypt32.CertStoreAddDisposition.CERT_STORE_ADD_ALWAYS, IntPtr.Zero))
+            using (SafeCertContextHandle certContext = certificatePal.GetCertContext())
             {
-                Exception e = Marshal.GetHRForLastWin32Error().ToCryptographicException();
-                certStore?.Dispose();
-                throw e;
+                if (certStore.IsInvalid ||
+                    !Interop.Crypt32.CertAddCertificateLinkToStore(certStore, certContext, Interop.Crypt32.CertStoreAddDisposition.CERT_STORE_ADD_ALWAYS, IntPtr.Zero))
+                {
+                    Exception e = Marshal.GetHRForLastWin32Error().ToCryptographicException();
+                    certStore?.Dispose();
```

Comment:
certStore was already dereferenced, so it isn't null.

```suggestion
                    certStore.Dispose();
```